usersession/userd/privileged_desktop_launcher_internal_test.go

```diff
@@ -0,0 +1,486 @@
+Name[kw]=Egry fenester noweth en modh privedh
+Name[ms]=Buka Tetingkap Baru dalam mod menyamar
+Name[nl]=Nieuw venster openen in incognito-modus
+Name[pt_BR]=Abrir uma nova janela em modo anônimo
+Name[ro]=Deschide o fereastră nouă în mod incognito
+Name[ru]=Открыть новое окно в режиме инкогнито
+Name[sl]=Odpri novo okno v načinu brez beleženja
+Name[sv]=Öppna ett nytt inkognitofönster
+Name[ug]=يوشۇرۇن ھالەتتە يېڭى كۆزنەك ئاچ
+Name[uk]=Відкрити нове вікно у приватному режимі
+Name[vi]=Mở cửa sổ mới trong chế độ ẩn danh
+Name[zh_CN]=以隐身模式打开新窗口
+Name[zh_TW]=以匿名模式開啟新視窗
+Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/chromium_chromium.desktop /snap/bin/chromium --incognito
+
+[Desktop Action TempProfile]
+Name=Open a New Window with a temporary profile
+Name[ast]=Abrir una ventana nueva con perfil temporal
+Name[bg]=Отваряне на Нов прозорец с временен профил
+Name[bn]=সাময়িক প্রোফাইল সহ একটি নতুন উইন্ডো খুলুন
+Name[bs]=Otvori novi prozor pomoću privremenog profila
+Name[ca]=Obre una finestra nova amb un perfil temporal
+Name[ca@valencia]=Obri una finestra nova amb un perfil temporal
+Name[de]=Ein neues Fenster mit einem temporären Profil öffnen
+Name[en_AU]=Open a New Window with a temporary profile
+Name[eo]=Malfermi novan fenestron portempe
+Name[es]=Abrir una ventana nueva con perfil temporal
+Name[et]=Ava uus aken ajutise profiiliga
+Name[eu]=Ireki leiho berria behin-behineko profil batekin
+Name[fi]=Avaa uusi ikkuna käyttäen väliaikaista profiilia
+Name[fr]=Ouvrir une nouvelle fenêtre avec un profil temporaire
+Name[gl]=Abrir unha nova xanela con perfil temporal
+Name[he]=פתיחת חלון חדש עם פרופיל זמני
+Name[hy]=Բացել նոր պատուհան ժամանակավոր հատկագրով
+Name[ia]=Aperi un nove fenestra con un profilo provisori
+Name[it]=Apri una nuova finestra con un profilo temporaneo
+Name[ja]=一時プロファイルで新しいウィンドウを開く
+Name[ka]=ახალი ფანჯრის გახსნა დროებით პროფილში
+Name[kw]=Egery fenester noweth gen profil dres prys
+Name[ms]=Buka Tetingkap Baru dengan profil sementara
+Name[nb]=Åpne et nytt vindu med en midlertidig profil
+Name[nl]=Nieuw venster openen met een tijdelijk profiel
+Name[pt_BR]=Abrir uma nova janela com um perfil temporário
+Name[ro]=Deschide o fereastră nouă cu un profil temporar
+Name[ru]=Открыть новое окно с временным профилем
+Name[sl]=Odpri novo okno z začasnim profilom
+Name[sv]=Öppna ett nytt fönster med temporär profil
+Name[ug]=ۋاقىتلىق سەپلىمە ھۆججەت بىلەن يېڭى كۆزنەك ئاچ
+Name[vi]=Mở cửa sổ mới với hồ sơ tạm
+Name[zh_CN]=以临时配置文件打开新窗口
+Name[zh_TW]=以暫時性個人身分開啟新視窗
+Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/chromium_chromium.desktop /snap/bin/chromium --temp-profile
+`
+
+func existsOnMockFileSystem(desktop_file string) bool {
+	return strutil.ListContains(mockFileSystem, desktop_file)
+}
+
+func (s *privilegedDesktopLauncherInternalSuite) TestDesktopFileIDToFilenameSucceedsWithValidId(c *C) {
+	restore := userd.MockRegularFileExists(existsOnMockFileSystem)
```

Comment:
we also need a happy test version where we don't mock the RegularFileExists function, I'm happy to write this for you